### ISSUE TITLE
Output layer for time series classification with Keras- which layer to use for multistep classification?

### ISSUE THREAD

**Issue by anaalvarezmari** [keras-team/keras#14598]
Hi Keras team
I wonder if you could help me , I have a 3 class time series classification problem, for which I need to predict the class up to two steps in the future. I have coded the output with one hot encoder so I have 3 columns I have doubts on the dimension of the output layer. When I worked with just one step to the future I could use

"model.add(Dense(3,activation='softmax'))"

and the the code I have worked ok, but I am not sure what output layer to use instead when I have 2 steps.

The dimensions of my training data are as follows:

ytrain2 (100861, 2, 3) xtrain2 (100861, 6, 9)

The bit of my code that does the training is as follows

from numpy import array
print(xnorm.shape)
Xtrain2, ytrain2 = split_sequence(xnorm,ynorm,6,2)
print(ytrain2)
Xtestf, ytestf = split_sequence(xtestnorm,ytestnorm,6,2)
print(ytrain2.shape)
recallV=list()
misclasV=list()
for i in n:
  for j in neurons:
    from keras.layers import Dense
    from keras.models import Sequential
    from keras.layers import Dropout
    import tensorflow as tf
    import keras as ks
    n_input = Xtrain2.shape[1] * Xtrain2.shape[2]
    ny=ytrain2.shape[1] * ytrain2.shape[2]
    X = Xtrain2.reshape(Xtrain2.shape[0],n_input)
    ytrain3=ytrain2.reshape(ytrain2.shape[0], ny)
    optimizer = ks.optimizers.Adam(lr=0.01)
    from keras.callbacks import ReduceLROnPlateau
    from keras.callbacks import EarlyStopping
    callbackEarlyStop = tf.keras.callbacks.EarlyStopping(monitor='val_loss', patience=15)

    reduce_lr = ReduceLROnPlateau(monitor='val_loss', factor=0.95,
                              patience=5, min_lr=0.0000001,verbose=1)
    # define model
    model = Sequential()
    model.add(Dense(j, activation='relu', input_dim=n_input))
    model.add(Dropout(0.1))
    model.add(Dense(j, activation='relu'))
    model.add(Dropout(0.1))
    ##this is what I need to change for something that makes sense
    **model.add(Dense(3,activation='softmax'))**
    model.compile(optimizer=optimizer, loss='categorical_crossentropy',metrics=['accuracy'])

    model.fit(X, ytrain3,epochs=100,batch_size=256,shuffle=False,validation_split=0.2,callbacks=[reduce_lr,callbackEarlyStop], verbose=2)

**Comment by jvishnuvardhan** [comment]
@anaalvarezmari Sorry for the late response. Is this still an issue for you?

This repo is mainly focussed on bugs/performance related issues. If you think this is more like a support question, then please post it in [TF Forum](https://discuss.tensorflow.org/) where there is a large community supporting each other and learning from each other.

If you think this is a bug, then please provide a simple standalone code to reproduce the issue. Thanks!
